src/solvers/prop/cover_goals.h

```diff
@@ -71,10 +73,9 @@ class cover_goalst
 
   // managing the goals
 
-  void add(const literalt condition)
+  void add(exprt condition)
   {
-    goals.push_back(goalt());
-    goals.back().condition=condition;
+    goals.emplace_back(condition);
```

Comment:
I believe this requires `std::move(condition)` to make use of the pass-by-value. (Else I'd suggest to pass a const reference.)

Reply:
ok